dist/birthdetails/birthdetails.js

```diff
@@ -0,0 +1,407 @@
+                    year: this.birthData.year,
+                    tobhr: this.birthData.hour,
+                    tobmin:this.birthData.minute,
+                    tobam: this.birthData.ampm,
+                    unsure: (this.birthData.unsure)?'on':'off',
+                    AutocompleteTarget: (this.birthData.countryShort === "USA")?this.birthData.usState:this.birthData.countryShort,
+                    Contact0City3: this.birthData.city,
+                    Contact0_birthplaceid:this.birthData.birthPlaceId
+                };
+                let formData = new FormData();
+                for (let birthPayloadKey in _birthPayload){
+                    formData.append(birthPayloadKey,_birthPayload[birthPayloadKey]);
+                }
+                fetch(`https://services.astrologyanswers.com/aa/update_birth_details`,
+                    {
+                        method: "POST",
+                        body: formData
+                    }).then(response => {
+                    if (response.status === 200) {
+                        response.json().then(data => {
+                            if (data['status'] === 'success') {
+                                window.location.href = this.nextUrl;
+                                return;
+                            }
+                        });
+                    }
+                    else {
+                        console.log('Error in birth update api : ' + response.status);
+                        this.cities=[];
+                    }
+                });
+
+            }
+        },
+        validateBirthDetails() {
+            this.birthDetailErrors.splice(0,this.birthDetailErrors.length);
+            if(this.user.firstname === '') {
+                this.birthDetailErrors.push('Please enter First Name');
+            }
+            if(this.birthData.year === '' || this.birthData.month === '' || this.birthData.day === '') {
+                this.birthDetailErrors.push('Please enter Date of Birth');
+            }
+            if(this.birthData.countryShort === '') {
+                this.birthDetailErrors.push('Please Select Country');
+            }else if(this.birthData.countryShort === 'USA' && this.birthData.usState === '') {
+                this.birthDetailErrors.push('Please Select State');
+            }
+            if(this.birthData.city === ''){
+                this.birthDetailErrors.push('Please Select City');
+            }
+            return this.birthDetailErrors.length === 0;
+        },
+        fetchBirthDetails(_nextUrl){
+            if(_nextUrl === '' || typeof _nextUrl === 'undefined'){
+                console.log('NextUrl required');
+                return;
+            }
+            const _urlParams = new URLSearchParams(window.location.search);
+            this.user.email = _urlParams.get('email') ?? '';
+            if(this.user.email === ''){
+                console.log('email required');
```

Comment:
```suggestion
                console.error('email required');
```
we have code verifying if we have email and nextUrl set or not, at 2 different places
should we look into getting it in 1 place?